### PR TITLE
Added header attribute to print alongside links and heading.

### PR DESCRIPTION
**Pull request by rickeywang**
For #31 

Introduce a new attribute called `header`, which is displayed alongside the UID and Links for MD/HTML. Gives more context for the Links, and allow for future stylistic changes to the published documents.

### CODE REVIEW

File: doorstop/core/item.py [hunk 1]
```diff
@@ -208,6 +212,11 @@ def data(self):
                 value = value.yaml
             elif key == 'text':
                 value = value.yaml
+            elif key == 'header':
+                if hasattr(value, 'yaml'):
```
Comment (rickeywang) [inline doorstop/core/item.py]: Handle for cases where header is empty

**Comment by jacebrowning** [comment]
This looks like a good start. I think we want this feature to be truly optional (at least to start) given that nobody else has asked for it thus far. This means that all the existing tests should pass without modification.

After that, you'll want to create an integration test for publishing a document containing items with headings.

**Comment by jacebrowning** [comment]
This is looking pretty good!

Maybe next you'd like to add a CLI option to enable this feature and use this to enable the flag in `cli/utilities/configure_settings`? Or perhaps the `ENABLE_HEADERS` value is simply set via `bool(os.getenv('DOORSTOP_ENABLE_HEADERS'))`.

**Comment by rickeywang** [comment]
The `header` attribute is exported for all other formats as well if the flag is enabled. 

I think it is redundant at this point to have all the tests duplicated for exporting to md, csv, xlsx, etc.. eventually the `header` attribute should become part of the yaml by default. Then just need a flag  to control how it is Published.

**Comment by jacebrowning** [comment]
@rickeywang Thanks for putting in all this work! Would you like to make any more changes before I ship this in a beta release?

**Comment by rickeywang** [comment]
No problem! Looks good now.

**Comment by jacebrowning** [comment]
This is released as `doorstop==1.5b1`.

**Pull request merged**